### PR TITLE
Inheritance support java client okhttp gson

### PR DESCRIPTION
**Pull request by konopski**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

Inheritance support for okhttp with gson parser.
@cbornet  Can you please take a look?

### CODE REVIEW

**Comment by wing328** [comment]
Similar work has been done before: https://github.com/swagger-api/swagger-codegen/pull/4729

cc @bbdouglas @JFCote @sreeshas @jfiala @lukoyanov @cbornet

File: samples/client/petstore/java/okhttp-gson/src/main/java/io/swagger/client/JSON.java [hunk 1]
```diff
@@ -40,8 +47,42 @@
...
+    public static GsonBuilder createGson() {
+        GsonFireBuilder fireBuilder = new GsonFireBuilder()
+          .registerTypeSelector(Animal.class, new TypeSelector() {
+            @Override
+            public Class getClassForElement(JsonElement readElement) {
+                Map classByDiscriminatorValue = new HashMap();
+                classByDiscriminatorValue.put("Cat".toUpperCase(), Cat.class);
+                classByDiscriminatorValue.put("Dog".toUpperCase(), Dog.class);
+                classByDiscriminatorValue.put("Animal".toUpperCase(), Animal.class);
+                               return getClassByDiscriminator(
```
Comment (cbornet) [inline samples/client/petstore/java/okhttp-gson/src/main/java/io/swagger/client/JSON.java]: Alignment issue

File: samples/client/petstore/java/retrofit2/src/main/java/io/swagger/client/model/AdditionalPropertiesClass.java [hunk 2]
```diff
@@ -130,6 +130,6 @@ private String toIndentedString(java.lang.Object o) {
     }
     return o.toString().replace("\n", "\n    ");
   }
-
+
```
Comment (cbornet) [inline samples/client/petstore/java/retrofit2/src/main/java/io/swagger/client/model/AdditionalPropertiesClass.java]: Can we avoid this unwanted space ?

**Comment by konopski** [comment]
Sorry, this PR is getting messy with some old merges and changes from previous branches with retrofit. 
I will clean up changes and ask for revision.

**Comment by konopski** [comment]
I cleaned up the changes, including your remarks.
No idea why travis did not pass and how to reproduce the problem locally.

**Comment by lukoyanov** [comment]
> No idea why travis did not pass and how to reproduce the problem locally.
Travis build failure is more than likely not related to changes in this PR. It has some issues.

**Comment by wing328** [comment]
Travis test passed via https://travis-ci.org/swagger-api/swagger-codegen/builds/305624150

**Pull request merged**